Update CIS scan docs

catherineluse opened the pull request:
I'm going to merge this PR to make it easier to review at this URL: https://staging.rancher.com/docs/rancher/v2.x/en/cis-scans/. Feedback should be left on this PR.

prachidamle commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -7,19 +7,46 @@ _Available as of v2.4.0_
 Rancher can run a security scan to check whether Kubernetes is deployed according to security best practices as defined in the CIS Kubernetes Benchmark.
 
-The Center for Internet Security (CIS) is a 501(c)(3) nonprofit organization, formed in October 2000, with a mission is to "identify, develop, validate, promote, and sustain best practice solutions for cyber defense and build and lead communities to enable an environment of trust in cyberspace". The organization is headquartered in East Greenbush, New York, with members including large corporatio ...
+The `rancher-cis-application` leverages [kube-bench,](https://github.com/aquasecurity/kube-bench) an open-source tool from Aqua Security, to check clusters for CIS Kubernetes Benchmark compliance. Also, to generate a cluster-wide report, the application utilizes [Sonobuoy](https://github.com/vmware-tanzu/sonobuoy) for report aggregation.
```
It should be `rancher-cis-benchmark`app  instead of `rancher-cis-application`

catherineluse replied:
updated

prachidamle commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -7,19 +7,46 @@ _Available as of v2.4.0_
+- EKS
+- GKE
+
+EKS and GKE have their own CIS Benchmarks published by `kube-bench`. The corresponding test profiles are used by default for those clusters.
```
Maybe also mention that for rest of cluster types, Generic CIS 1.5 (default) profile will be used.

catherineluse replied:
updated

prachidamle commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -7,19 +7,46 @@ _Available as of v2.4.0_
+
+The `rancher-cis-benchmark` currently supports the CIS 1.5 Benchmark version.
+
+> **Note:** The old and new versions of monitoring shouldn't run on the cluster at the same time.
```
The Note here mentions monitoring - its should be changed to say: The old and new versions of  rancher-cis-benchmark shouldn't run on the cluster at the same time.

catherineluse replied:
updated

prachidamle commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -29,7 +56,113 @@ CIS Benchmarks are best practices for the secure configuration of a target syste
+When a ClusterScan custom resource is created, it launches a new CIS scan on the cluster for the chosen ClusterScanProfile.
+
+Note: There is currently a limitation of running only one CIS scan at a time for a cluster.
+
```
You can add: If you create multiple ClusterScan custom resources, they will be run one after the other by the Operator and until one scan finishes the rest of the ClusterScan custom resources will be in "Pending" state.

catherineluse replied:
updated

prachidamle commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -29,7 +56,113 @@ CIS Benchmarks are best practices for the secure configuration of a target syste
+
+1. Go to the **Cluster Explorer** in the Rancher UI. In the top left dropdown menu, click **Cluster Explorer > CIS Benchmark.**
+1. In the **Scans** section, click **Create.**
+1. Choose a cluster scan profile. The profile determines which CIS Benchmark version will be used and which tests will be performed. The CIS Operator will choose a default profile applicable to the type of Kubernetes cluster it is installed on.
```
The second line here should say: "If you choose Default profile then the CIS Operator will choose a profile applicable to the type of Kubernetes cluster it is installed on."

catherineluse replied:
updated

prachidamle commented on content/rancher/v2.x/en/cis-scans/skipped-tests/_index.md:
```diff
@@ -0,0 +1,59 @@
+
+# CIS Benchmark v1.5
+
+This skipped and not applicable tests for CIS Benchmark v1.5 are listed in [this file.](https://github.com/rancher/kontainer-driver-metadata/blob/dev-v2.5/rke/cis_config.go#L81-L119)
```
No we should not give link to this file in the docs, instead we need to list the TestIds as we have done below for CIS Benchmark v1.4 Skipped Tests

catherineluse replied:
updated

prachidamle commented on content/rancher/v2.x/en/cis-scans/rbac/_index.md:
```diff
@@ -1,5 +1,6 @@
 ---
-title: RBAC
+title: Roles-based Access Contro
```
title has typo "Roles-based Access Contro"

prachidamle commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -29,7 +56,113 @@ CIS Benchmarks are best practices for the secure configuration of a target syste
+To view the generated CIS scan reports,
+
+1. In the **Cluster Explorer,** go to the top left dropdown menu and click **Cluster Explorer > CIS Benchmark.**
+1. The **Scans** page will show the generated reports. To see a detailed report, go to a scan report and click the name.
```
Also we can mention that one can download the report - either from the Scans list or from the scan detail page.

catherineluse replied:
updated

prachidamle commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -29,7 +56,113 @@ CIS Benchmarks are best practices for the secure configuration of a target syste
+1. The **Scans** page will show the generated reports. To see a detailed report, go to a scan report and click the name.
 
 # About the Generated Report
 
```
This "About the Generated Report" section needs to be updated to mention CIS 1.5  instead of CIS 1.4 and rancher 2.5 instead of 2.4

catherineluse replied:
updated

sowmyav27 commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -29,7 +56,113 @@ CIS Benchmarks are best practices for the secure configuration of a target syste
+1. In the **Cluster Explorer,** go to the top left dropdown menu and click **Cluster Explorer > CIS Benchmark.**
+1. The **Scans** page will show the generated reports. To see a detailed report, go to a scan report and click the name.
 
 # About the Generated Report
```
Also columns in the generated report have changed

catherineluse replied:
I didn't intend to mention Rancher 2.4. I've changed it to:

> In Rancher v2.5, the scan will use the CIS Benchmark v1.5.

catherineluse replied:
To update the documentation of the columns in the report, I deleted the columns that aren't in the CIS v2 report.

But there are some columns that are in the v2 report but not the v1 report. What should be the descriptions of these:

- audit
- audit_config
- test_info
- commands
- config_commands
- actual_value
- expected_result

@prachidamle

sowmyav27 commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -54,66 +187,45 @@ The report contains the following information:
+
+# About Skipped and Not Applicable Tests
+
+For a list of skipped and not applicable tests, refer to [this page.](./skipped-tests)
```
This page lists the tests which are skipped now, but in v2 CIS this is not the case.
Can we have a list called out separately for CIS v1 and CIS v2 for 1.15 cluster scan ?

catherineluse replied:
I created a new page in the CIS scan docs for Rancher v2.4. Because Rancher 2.4 let you use CIS Benchmark 1.5 and 1.4, I put both lists there. Then in the 2.5 docs, I made it so that only the skipped and not applicable tests for CIS Benchmark 1.5 are listed. Is that OK?

sowmyav27 commented on content/rancher/v2.x/en/cis-scans/rbac/_index.md:
```diff
@@ -1,5 +1,6 @@
 ---
-title: RBAC
+title: Roles-based Access Contro
```
Missing an `l`

catherineluse replied:
updated

catherineluse commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -7,19 +7,46 @@ _Available as of v2.4.0_
 Rancher can run a security scan to check whether Kubernetes is deployed according to security best practices as defined in the CIS Kubernetes Benchmark.
 
-The Center for Internet Security (CIS) is a 501(c)(3) nonprofit organization, formed in October 2000, with a mission is to "identify, develop, validate, promote, and sustain best practice solutions for cyber defense and build and lead communities to enable an environment of trust in cyberspace". The organization is headquartered in East Greenbush, New York, with members including large corporatio ...
+The `rancher-cis-application` leverages [kube-bench,](https://github.com/aquasecurity/kube-bench) an open-source tool from Aqua Security, to check clusters for CIS Kubernetes Benchmark compliance. Also, to generate a cluster-wide report, the application utilizes [Sonobuoy](https://github.com/vmware-tanzu/sonobuoy) for report aggregation.
```
yes, updated

catherineluse commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -7,19 +7,46 @@ _Available as of v2.4.0_
 
+More test profiles were added. In Rancher v2.4, permissive and hardened profiles were included. In Rancher v2.5, the following profiles are available:
+
+- Generic CIS 1.5 (default)
```
updated:

> 
> - Generic CIS 1.5
> - RKE permissive
> - RKE hardened
> - EKS
> - GKE
> 
> The default profile depends on the type of cluster that will be scanned:
> 
> - For RKE Kubernetes clusters, the RKE permissive profile is the default.
> - EKS and GKE have their own CIS Benchmarks published by `kube-bench`. The corresponding test profiles are used by default for those clusters.
> - For cluster types other than RKE, EKS and GKE, the Generic CIS 1.5 profile will be used by default.

catherineluse commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -7,19 +7,46 @@ _Available as of v2.4.0_
+
+The `rancher-cis-benchmark` currently supports the CIS 1.5 Benchmark version.
+
+> **Note:** The old and new versions of monitoring shouldn't run on the cluster at the same time.
```
updated

catherineluse commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -29,7 +56,113 @@ CIS Benchmarks are best practices for the secure configuration of a target syste
 
 The official Benchmark documents are available through the CIS website. The sign-up form to access the documents is [here.](https://learn.cisecurity.org/benchmarks)
```
updated

catherineluse commented on content/rancher/v2.x/en/cis-scans/_index.md:
```diff
@@ -29,7 +56,113 @@ CIS Benchmarks are best practices for the secure configuration of a target syste
+- `rancher-cis-benchmark-crds`, the custom resource definition chart
+- `rancher-cis-benchmark`, the chart deploying [rancher/cis-operator](https://github.com/rancher/cis-operator)
+
+To install the charts, run the following commands:
```
updated

catherineluse commented on content/rancher/v2.x/en/cis-scans/_index.md:
````diff
@@ -29,7 +56,113 @@ CIS Benchmarks are best practices for the secure configuration of a target syste
+
+Run the following commands:
+
+```
````
updated

catherineluse commented:
@prachidamle and @sowmyav27 I have addressed your feedback in this PR https://github.com/rancher/docs/pull/2731

I added tables to list the skipped and not applicable tests in the Benchmark 1.5. The new tables are here: https://staging.rancher.com/docs/rancher/v2.x/en/cis-scans/skipped-tests/ And I moved the Benchmark 1.4 tables to the docs for CIS scans in Rancher v2.4: https://staging.rancher.com/docs/rancher/v2.x/en/cis-scans/legacy/skipped-tests/

To get the reason for skipping the tests, I looked in the Go code. To get the description of the tests, I looked in the Benchmark 1.5.1

@prachidamle was it ok to use the Benchmark 1.5.1? I tried to get the Benchmark 1.5 but it wasn't available for download from the CIS website:

<img width="759" alt="Screen Shot 2020-10-03 at 3 50 11 PM" src="https://user-images.githubusercontent.com/20599230/95006264-79781900-05b7-11eb-848c-aae767b9b78b.png">

And I noticed that test ID 5.6.4 had no description in the benchmark, because that whole section is missing in the benchmark. You can see it goes from 5.5 to 5.7:
<img width="789" alt="Screen Shot 2020-10-03 at 3 45 25 PM" src="https://user-images.githubusercontent.com/20599230/95006276-aaf0e480-05b7-11eb-9bc2-b0e8a0a62fa3.png">

I assumed that was a numbering error in the benchmark itself, so I used the description for 5.7.4 instead.

And I have some outstanding questions. The first question is, how should we describe these fields in the generated report?

- audit
- audit_config
- test_info
- commands
- config_commands
- actual_value
- expected_result

The second question is, how do the docs need to change based on the feedback on the RBAC page? https://github.com/rancher/docs/pull/2712#pullrequestreview-501586864 Do we just need to say that in Rancher, you only have permission to create scans and view reports if you are the cluster owner? And is it possible to give someone permission to use the CIS scan through the Rancher UI?

sowmyav27 commented:
@catherineluse I see 5.6 section in CIS 1.5.0 document. 
<img width="608" alt="Screen Shot 2020-10-04 at 3 25 19 PM" src="https://user-images.githubusercontent.com/26032343/95028571-d75b3e00-0655-11eb-94db-d4b3e640bf7e.png">

prachidamle commented:
@catherineluse 
- I have forwarded the CIS 1.5.0 pdf
- For the below fields added to the report:

audit - This is the audit check that kube-bench runs for this test
audit_config - any config applicable to the audit script
test_info - test related info as reported by kube-bench if any
commands - test related commands as reported by kube-bench if any
config_commands - test related config data as reported by kube-bench if any
actual_value - tests actual value present if reported by kube-bench
expected_result - tests expected result present if reported by kube-bench

- RBAC review - @cbron pls can you comment on this https://github.com/rancher/docs/pull/2712#discussion_r499791385 if the docs look good.

catherineluse commented:
I have responded to feedback here https://github.com/rancher/docs/pull/2735
